main.go

```diff
@@ -0,0 +1,312 @@
+	flag.StringVar(&opts.targetkubeconfig, "target-kubeconfig", "", "kubeconfig of target cluster where objects will be synced to")
+
+	zapOpts := kubezap.Options{
+		Development: true,
+	}
+	zapOpts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	if opts.kcpkubeconfig == "" || opts.targetkubeconfig == "" {
+		return errors.New("both --kcp-kubeconfig and --target-kubeconfig must be specified")
+	}
+
+	logger := kubezap.New(kubezap.UseFlagOptions(&zapOpts))
+	ctrl.SetLogger(logger)
+	log := logger.WithName("setup")
+
+	// setup downstream reconciliation
+	kcpConfig, err := clientcmd.BuildConfigFromFlags("", opts.kcpkubeconfig)
+	if err != nil {
+		return err
+	}
+
+	provider, err := apiexport.New(kcpConfig, apiexport.Options{})
+	if err != nil {
+		return err
+	}
+
+	mgr, err := mcmanager.New(kcpConfig, provider, manager.Options{
+		Logger: logger.WithName("manager"),
+	})
+	if err != nil {
+		return err
+	}
+
+	// setup upstream reconciliation
+	targetConfig, err := clientcmd.BuildConfigFromFlags("", opts.targetkubeconfig)
+	if err != nil {
+		return err
+	}
+	targetClient, err := ctrlruntimeclient.New(targetConfig, ctrlruntimeclient.Options{})
+	if err != nil {
+		return err
+	}
+
+	// build the reconciler
+	rec := &reconciler{
+		log:           &logger,
+		targetClient:  targetClient,
+		clusterGetter: mgr.GetCluster,
+	}
+	if err := rec.SetupWithManager(mgr, targetConfig, log); err != nil {
+		return err
+	}
+	log.Info("Setup complete")
+
+	log.Info("Starting provider and manager")
+	g, ctx := errgroup.WithContext(context.Background())
+	g.Go(func() error { return wrapError("provider", provider.Run(ctx, mgr)) })
+	g.Go(func() error { return wrapError("manager", mgr.Start(ctx)) })
+
+	return g.Wait()
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, req mcreconcile.Request) (reconcile.Result, error) {
+	clustername := req.ClusterName
+	log := r.log.
+		WithName("reconciler").
+		WithValues("clustername", clustername).
+		WithValues("object", req.NamespacedName.String())
+
+	log.Info("Starting Reconciliation")
+
+	cluster, err := r.clusterGetter(ctx, clustername)
+	if err != nil {
+		log.Error(err, "failed to get cluster")
+		return reconcile.Result{}, err
+	}
+	kcpclient := cluster.GetClient()
+
+	up := &mongodbv1.MongoDBCommunity{}
+	upErr := kcpclient.Get(ctx, req.NamespacedName, up)
+	if ctrlruntimeclient.IgnoreNotFound(upErr) != nil {
+		log.Error(err, "failed to get mongodb from kcp")
+		return reconcile.Result{}, err
+	}
+	upExists := !apierrors.IsNotFound(upErr)
+
+	down := &mongodbv1.MongoDBCommunity{}
+	downErr := r.targetClient.Get(ctx, req.NamespacedName, down)
+	if ctrlruntimeclient.IgnoreNotFound(downErr) != nil {
+		log.Error(err, "failed to get mongodb from downstream cluster")
+		return reconcile.Result{}, err
+	}
+	downExists := !apierrors.IsNotFound(downErr)
+
+	// if upstream does not exist, make sure downstream also does not exist
+	if !upExists && downExists {
```

Comment:
```suggestion
	if !upExists {
		if !downExists {
			return reconcile.Result{}, nil
		}
```

Otherwise if `!upExists && !downExists` this falls through until the `cmp.Diff` and further.